### PR TITLE
Add password reset typed events and modernize LostController

### PR DESCRIPTION
**Pull request by tcitworld**

### CODE REVIEW

File: core/Controller/LostController.php [hunk 1]
```diff
@@ -67,30 +71,19 @@
...
-	private $twoFactorManager;
-	/** @var IInitialStateService */
-	private $initialStateService;
-	/** @var IVerificationToken */
-	private $verificationToken;
+	protected IURLGenerator $urlGenerator;
+	protected IUserManager $userManager;
+	protected Defaults $defaults;
+	protected IL10N $l10n;
+	protected string $from;
```
[the remark this answers is not in the document]
Comment (tcitworld) [reply core/Controller/LostController.php:78]: The value comes from the `defaultMailAddress` service.
https://github.com/nextcloud/server/blob/b40481e8e5dfe30ef9568758490af675713ee0c2/core/Application.php#L70-L72
https://github.com/nextcloud/server/blob/b40481e8e5dfe30ef9568758490af675713ee0c2/lib/public/Util.php#L330-L360

So it should be safe. The value in constructor needs to be typed too though.

File: core/Events/PasswordResetEvent.php [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+ */
+namespace OC\Core\Events;
+
+use OCP\EventDispatcher\Event;
+use OCP\IUser;
+
+/**
+ * Emitted after the user password is reset.
+ *
+ * @since 24.0.0
```
Comment (PVince81) [inline core/Events/PasswordResetEvent.php:35]: 25 now if we merge soon

**Review by PVince81 — APPROVED** [review]
:+1: code looks fine to me, please adjust the `@since` for 25

**Comment by tcitworld** [comment]
Adjusted and rebased.

**Review by come-nc — APPROVED** [review]

**Pull request merged**